src/base/http/connection.h

```diff
@@ -27,41 +28,61 @@
  * exception statement from your version.
  */
 
-
 #pragma once
 
 #include <QElapsedTimer>
 #include <QObject>
 
+#include "types.h"
+
 class QTcpSocket;
 
 namespace Http
 {
-    class IRequestHandler;
-    struct Response;
-
     class Connection : public QObject
     {
         Q_OBJECT
         Q_DISABLE_COPY(Connection)
 
     public:
-        Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObject *parent = nullptr);
-        ~Connection();
+        // Connection takes ownership of socket
+        explicit Connection(QTcpSocket *socket, QObject *parent = nullptr);
+
+        void sendStatus(const ResponseStatus &status);
+        void sendHeaders(const HeaderMap &headers);
+        void sendContent(const QByteArray &data);
+        void close();
 
-        bool hasExpired(qint64 timeout) const;
-        bool isClosed() const;
+        Http::Environment enviroment() const;
+        qint64 inactivityTime() const;
+        qint64 bytesToWrite() const;
+
+    signals:
+        void bytesWritten(qint64 bytes);
+        void disconnected();
+        void readyRequest(Request request);
 
     private slots:
         void read();
 
     private:
-        static bool acceptsGzipEncoding(QString codings);
-        void sendResponse(const Response &response) const;
+        enum Status
+        {
+            PendingRequest,
+            PendingStatus,
+            PendingHeaders,
+            PendingContent,
+            WritingContent,
+            Closed
+        };
 
         QTcpSocket *m_socket;
-        IRequestHandler *m_requestHandler;
-        QByteArray m_receivedData;
         QElapsedTimer m_idleTimer;
+        QByteArray m_receivedData;
+        QByteArray m_statusHeaderBuffer;
+        Request m_request;
+        Status m_status = PendingRequest;
+        qint64 m_pendingContentSize = 0;
+        bool m_persistantConnection = true;
```

Comment:
My English is not so good, but at least it is more familiar to use the word "persist**e**nt".